### PR TITLE
New Resource - `azurerm_signalr_service_custom_certificate`

### PR DESCRIPTION
**Pull request by xiaxyi**
acc test:
--- PASS: TestAccCustomCertBindingSignalrService_basic (619.70s)
PASS
ok      github.com/hashicorp/terraform-provider-azurerm/internal/services/signalr       620.378s

### CODE REVIEW

File: internal/services/signalr/signalr_custom_certificate_binding_resource.go [hunk 1]
```diff
@@ -0,0 +1,282 @@
...
+				Pending:    []string{string(signalr.ProvisioningStateCreating), string(signalr.ProvisioningStateUpdating), string(signalr.ProvisioningStateFailed)},
+				Target:     []string{string(signalr.ProvisioningStateSucceeded)},
+				Refresh:    signalrCustomCertBindingStateRefreshFunc(ctx, client, id),
+				MinTimeout: 15 * time.Second,
+				Timeout:    30 * time.Minute,
+			}
+
+			if _, err := stateConf.WaitForStateContext(ctx); err != nil {
+				return fmt.Errorf("waiting for creation of %s: %+v", id, err)
+			}
```
Comment (catriona-m) [inline internal/services/signalr/signalr_custom_certificate_binding_resource.go:138]: would it make sense to use something like `client.CustomCertificatesCreateOrUpdateThenPoll` here?

File: internal/services/signalr/signalr_custom_certificate_binding_resource.go [hunk 2]
```diff
@@ -0,0 +1,282 @@
...
+			if props == nil {
+				return fmt.Errorf("retrieving %s: `model` was nil", *id)
+			}
+
+			if metadata.ResourceData.HasChange("certificate_version") {
+				existing.Model.Properties.KeyVaultSecretVersion = utils.String(signalRCustomCertBinding.CertificateVersion)
+			}
+
+			if err := client.CustomCertificatesCreateOrUpdateThenPoll(ctx, *id, *props); err != nil {
+				return fmt.Errorf("creating signalR custom certificate binding: %s: %+v", id, err)
```
Comment (catriona-m) [inline internal/services/signalr/signalr_custom_certificate_binding_resource.go:237]:
```suggestion
				return fmt.Errorf("updating signalR custom certificate binding: %s: %+v", id, err)
```

File: internal/services/signalr/signalr_custom_certificate_binding_resource.go [hunk 3]
```diff
@@ -0,0 +1,282 @@
...
+			return nil, "", fmt.Errorf("polling for %s: %+v", id, err)
+		}
+
+		if model := resp.Model; model != nil {
+			if model.Properties.ProvisioningState != nil {
+				return resp, string(*model.Properties.ProvisioningState), nil
+			}
+		}
+
+		return nil, "", fmt.Errorf("error fetching the custom certificate provisioing state")
```
Comment (catriona-m) [inline internal/services/signalr/signalr_custom_certificate_binding_resource.go:280]:
```suggestion
		return nil, "", fmt.Errorf("error fetching the custom certificate provisioning state")
```

File: internal/services/signalr/signalr_custom_certificate_binding_resource.go [hunk 4]
```diff
@@ -0,0 +1,282 @@
...
+	"github.com/hashicorp/go-azure-sdk/resource-manager/signalr/2023-02-01/signalr"
+	"github.com/hashicorp/terraform-provider-azurerm/internal/sdk"
+	keyVaultParse "github.com/hashicorp/terraform-provider-azurerm/internal/services/keyvault/parse"
+	keyVaultValidate "github.com/hashicorp/terraform-provider-azurerm/internal/services/keyvault/validate"
+	"github.com/hashicorp/terraform-provider-azurerm/internal/tf/pluginsdk"
+	"github.com/hashicorp/terraform-provider-azurerm/internal/tf/validation"
+	"github.com/hashicorp/terraform-provider-azurerm/utils"
+)
+
+type CustomCertBindingSignalrModel struct {
```
Comment (catriona-m) [inline internal/services/signalr/signalr_custom_certificate_binding_resource.go:18]:
```suggestion
type CustomCertBindingSignalrResourceModel struct {
```

File: internal/services/signalr/signalr_custom_certificate_binding_resource.go [hunk 5]
```diff
@@ -0,0 +1,282 @@
...
+			ValidateFunc: validation.Any(
+				keyVaultValidate.NestedItemId,
+				keyVaultValidate.NestedItemIdWithOptionalVersion,
+			),
+		},
+
+		"certificate_version": {
+			Type:         pluginsdk.TypeString,
+			Optional:     true,
+			Computed:     true,
```
Comment (catriona-m) [inline internal/services/signalr/signalr_custom_certificate_binding_resource.go:58]: does this need to be computed?

**Review by catriona-m — CHANGES_REQUESTED** [review]
Hi @xiaxyi, thanks for this. I've left a couple of comments inline for you to consider and then we can take another look at this. Thanks!

**Review by katbyte — APPROVED** [review]
LGTM 🍰

**Pull request merged**